src/ripple/overlay/impl/OverlayImpl.h

```diff
@@ -456,6 +456,44 @@ class OverlayImpl : public Overlay
 
     void
     sendEndpoints();
+
+private:
+    struct Stats
+    {
+
+        template <class Handler>
+        Stats (
+                Handler const& handler, 
+                beast::insight::Collector::ptr const& collector,
+                std::vector<beast::insight::Gauge>&& trafficGauges_)
+            : peerDisconnects (collector->make_gauge("Overlay","Peer_Disconnects"))
+            , trafficGauges (trafficGauges_)
+            , hook (collector->make_hook (handler))
+            { 
+            }
+            
+        beast::insight::Gauge peerDisconnects;
+        std::vector<beast::insight::Gauge> trafficGauges;
```

Comment:
I think this is better off as something like 
```
struct TrafficGauges {
    Gauge bytesIn;
    Gauge bytesOut;
    Gauge messagesIn;
    Gauge messagesOut;
};

std::vector<TrafficGauges> trafficGauges;
```

That way they're grouped w/ names instead of implicitly every 4 elements.